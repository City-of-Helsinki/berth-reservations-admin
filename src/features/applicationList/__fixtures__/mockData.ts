import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
