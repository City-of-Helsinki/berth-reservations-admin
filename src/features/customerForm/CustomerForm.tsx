import React from 'react';
import { useTranslation } from 'react-i18next';
import { Form, Formik } from 'formik';
import * as Yup from 'yup';
import { TFunction } from 'i18next';
import { TextArea, TextInput } from 'hds-react';

import { CustomerFormValues, FormProps } from './types';
import { CustomerGroup } from '../../@types/__generated__/globalTypes';
import Select from '../../common/select/Select';
import styles from './customerForm.module.scss';
import Button from '../../common/button/Button';
import { getCustomerGroupKey } from '../../common/utils/translations';
import Text from '../../common/text/Text';
import PrivateCustomerFields from './PrivateCustomerFields';
import OrganizationCustomerFields from './OrganizationCustomerFields';

export type CustomerFormProps = FormProps<CustomerFormValues>;

const getValidationSchema = (t: TFunction) => {
  return Yup.object<CustomerFormValues>().shape({
    address: Yup.string().required(t('forms.common.errors.required')),
    businessId: Yup.string().when('customerGroup', {
      is: (customerGroup) => customerGroup !== CustomerGroup.PRIVATE,
      then: (businessId: Yup.StringSchema) => businessId.required(t('forms.common.errors.required')),
    }),
    city: Yup.string().required(t('forms.common.errors.required')),
    customerGroup: Yup.string().oneOf(Object.values(CustomerGroup), t('forms.common.errors.required')),
    email: Yup.string().email(t('forms.common.errors.email')).required(t('forms.common.errors.required')),
    firstName: Yup.string().required(t('forms.common.errors.required')),
    lastName: Yup.string().required(t('forms.common.errors.required')),
    organizationName: Yup.string().when('customerGroup', {
      is: (customerGroup) => customerGroup !== CustomerGroup.PRIVATE,
      then: (organizationName: Yup.StringSchema) => organizationName.required(t('forms.common.errors.required')),
    }),
    postalCode: Yup.string().required(t('forms.common.errors.required')).length(5, t('forms.common.errors.postalCode')),
  });
};

const CustomerForm = ({ initialValues, isSubmitting, onSubmit, onCancel }: CustomerFormProps) => {
  const { t } = useTranslation();
  const validationSchema = getValidationSchema(t);

  const initial: CustomerFormValues = initialValues ?? {
    address: '',
    businessId: '',
    city: '',
    comment: '',
    customerGroup: CustomerGroup.PRIVATE,
    email: '',
    firstName: '',
    lastName: '',
    organizationName: '',
    phone: '',
    postalCode: '',
  };

  const customerGroupOptions = Object.keys(CustomerGroup).map((group) => ({
    label: t(getCustomerGroupKey(group as CustomerGroup)),
    value: group,
  }));

  return (
    <>
      <Text as="h4" color="brand">
        {t('forms.customer.title').toUpperCase()}
      </Text>
      <Formik
        initialValues={initial}
        onSubmit={(values) => onSubmit?.(values)}
        validationSchema={validationSchema}
        validateOnChange={false}
      >
        {({ errors, handleChange, values, validateForm }) => (
          <Form className={styles.form}>
            <Select
              id="customerGroup"
              value={values.customerGroup as string}
              options={customerGroupOptions}
              onChange={handleChange}
              label={t('forms.customer.customerGroup')}
              required
            />

            {values.customerGroup === CustomerGroup.PRIVATE ? (
              <PrivateCustomerFields values={values} handleChange={handleChange} errors={errors} />
            ) : (
              <OrganizationCustomerFields values={values} handleChange={handleChange} errors={errors} />
            )}

            <TextInput
              id="phone"
              label={t('forms.customer.phone')}
              value={values.phone || ''}
              onChange={handleChange}
              invalid={!!errors.phone}
              helperText={errors.phone}
            />
            <TextInput
              id="email"
              label={t('forms.customer.email')}
              value={values.email || ''}
              onChange={handleChange}
              invalid={!!errors.email}
              helperText={errors.email}
              required
            />
            <TextArea
              id="comment"
              label={t('forms.customer.comment')}
              value={values.comment || ''}
              onChange={handleChange}
              invalid={!!errors.comment}
              helperText={errors.comment}
            />

            <div className={styles.formActionButtons}>
              <Button variant="secondary" disabled={isSubmitting} color={'supplementary'} onClick={onCancel}>
                {t('forms.common.cancel')}
              </Button>
              <Button type="submit" disabled={isSubmitting} onClick={() => validateForm(values)}>
                {t('common.save')}
              </Button>
            </div>
          </Form>
        )}
      </Formik>
    </>
  );
};

export default CustomerForm;
