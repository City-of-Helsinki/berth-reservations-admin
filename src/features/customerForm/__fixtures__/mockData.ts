import { CUSTOMER_FORM_profile as PROFILE } from '../__generated__/CUSTOMER_FORM';
import { CustomerGroup, InvoicingType, Language } from '../../../@types/__generated__/globalTypes';
import { CustomerFormIdentifiers, CustomerFormValues } from '../types';

export const mockPrivateCustomerProfile: PROFILE = {
  __typename: 'ProfileNode',
  comment: '',
  customerGroup: CustomerGroup.PRIVATE,
  firstName: 'Risto Heikki',
  id: 'MOCK-PROFILE',
  language: Language.FINNISH,
  lastName: 'Ryti',
  organization: null,
  invoicingType: InvoicingType.ONLINE_PAYMENT,
  primaryAddress: {
    __typename: 'AddressNode',
    address: 'Mariankatu 2',
    city: 'Helsinki',
    id: 'MOCK-ADDRESS',
    postalCode: '00170',
  },
  primaryEmail: {
    __typename: 'EmailNode',
    email: 'test@example.com',
    id: 'MOCK-EMAIL',
  },
  primaryPhone: {
    __typename: 'PhoneNode',
    id: 'MOCK-PHONE',
    phone: '+358 00 000 0000',
  },
};

export const mockOrganizationCustomerProfile: PROFILE = {
  __typename: 'ProfileNode',
  comment: '',
  customerGroup: CustomerGroup.COMPANY,
  firstName: 'Ky√∂sti',
  id: 'MOCK-PROFILE',
  language: Language.FINNISH,
  lastName: 'Kallio',
  invoicingType: InvoicingType.ONLINE_PAYMENT,
  organization: {
    __typename: 'OrganizationNode',
    address: 'Mariankatu 2',
    businessId: '1234567-8',
    city: 'Helsinki',
    id: 'MOCK-ORGANIZATION',
    name: 'Suomi Oy',
    postalCode: '00170',
  },
  primaryAddress: null,
  primaryEmail: {
    __typename: 'EmailNode',
    email: 'test@example.com',
    id: 'MOCK-EMAIL',
  },
  primaryPhone: {
    __typename: 'PhoneNode',
    id: 'MOCK-PHONE',
    phone: '+358 00 000 0000',
  },
};

export const mockPrivateCustomerFormValues: CustomerFormValues = {
  address: 'Mariankatu 2',
  businessId: '',
  city: 'Helsinki',
  comment: '',
  customerGroup: CustomerGroup.PRIVATE,
  email: 'test@example.com',
  firstName: 'Pehr Evind',
  lastName: 'Svinhufvud',
  organizationName: '',
  phone: '+358 00 000 0000',
  postalCode: '00170',
};

export const mockPrivateCustomerFormIdentifiers: CustomerFormIdentifiers = {
  id: 'MOCK-PROFILE',
  initialCustomerGroup: CustomerGroup.PRIVATE,
  primaryAddressId: 'MOCK-ADDRESS',
  primaryEmailId: 'MOCK-EMAIL',
  primaryPhoneId: 'MOCK-PHONE',
};

export const mockOrganizationCustomerFormValues: CustomerFormValues = {
  address: 'Mariankatu 2',
  businessId: '1234567-8',
  city: 'Helsinki',
  comment: '',
  customerGroup: CustomerGroup.COMPANY,
  email: 'test@example.com',
  firstName: 'Lauri Kristian',
  lastName: 'Relander',
  organizationName: 'Suomi Oy',
  phone: '+358 00 000 0000',
  postalCode: '00170',
};

export const mockOrganizationCustomerFormIdentifiers: CustomerFormIdentifiers = {
  id: 'MOCK-PROFILE',
  initialCustomerGroup: CustomerGroup.COMPANY,
  primaryAddressId: 'MOCK-ADDRESS',
  primaryEmailId: 'MOCK-EMAIL',
  primaryPhoneId: 'MOCK-PHONE',
};
