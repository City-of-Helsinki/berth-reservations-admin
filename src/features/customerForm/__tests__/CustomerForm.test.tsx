import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import CustomerForm from '../CustomerForm';
import { mockOrganizationCustomerFormValues, mockPrivateCustomerFormValues } from '../__fixtures__/mockData';

describe('CustomerForm', () => {
  const getWrapper = (props: {}) => mount(<CustomerForm {...props} />);

  it('renders correctly with private customer profile', () => {
    const wrapper = getWrapper({ initialValues: mockPrivateCustomerFormValues });
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders correctly with organization customer profile', () => {
    const wrapper = getWrapper({ initialValues: mockOrganizationCustomerFormValues });
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('empty profile can not be submitted', async () => {
    const onSubmit = jest.fn();
    const wrapper = getWrapper({ onSubmit });
    await act(async () => {
      wrapper.find('form').simulate('submit');
    });
    expect(onSubmit.mock.calls.length).toEqual(0);
  });

  it('private customer profile can be submitted', async () => {
    const onSubmit = jest.fn();
    const wrapper = getWrapper({ initialValues: mockPrivateCustomerFormValues, onSubmit });
    await act(async () => {
      wrapper.find('form').simulate('submit');
    });
    expect(onSubmit.mock.calls.length).toEqual(1);
  });

  it('organization customer profile can be submitted', async () => {
    const onSubmit = jest.fn();
    const wrapper = getWrapper({ initialValues: mockOrganizationCustomerFormValues, onSubmit });
    await act(async () => {
      wrapper.find('form').simulate('submit');
    });
    expect(onSubmit.mock.calls.length).toEqual(1);
  });
});
