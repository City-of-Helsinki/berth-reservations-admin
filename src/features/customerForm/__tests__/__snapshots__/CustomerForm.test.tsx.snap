// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerForm renders correctly with organization customer profile 1`] = `
Array [
  <h4
    class="text brand h4"
  >
    ASIAKKAAN TIEDOT
  </h4>,
  <form
    action="#"
    class="form"
  >
    <div
      class="Select-module_root__Ka5uO select"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="customerGroup-toggle-button"
        id="customerGroup-label"
      >
        Asiakasryhmä
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="Select-module_wrapper__1WQXs"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="customerGroup-label customerGroup-toggle-button"
          aria-owns="customerGroup-menu"
          class="Select-module_button__1aIsm"
          id="customerGroup-toggle-button"
          type="button"
        >
          Yritys
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
        <ul
          aria-labelledby="customerGroup-label"
          aria-required="true"
          class="Select-module_menu__1H2aU"
          id="customerGroup-menu"
          role="listbox"
          style="max-height: 260px;"
          tabindex="-1"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="organizationName"
      >
        Yrityksen tai yhteisön nimi
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="organizationName"
          required=""
          type="text"
          value="Suomi Oy"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="businessId"
      >
        Y-tunnus
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="businessId"
          required=""
          type="text"
          value="1234567-8"
        />
      </div>
    </div>
    <hr
      class="divider"
    />
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="address"
      >
        Jakeluosoite
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="address"
          required=""
          type="text"
          value="Mariankatu 2"
        />
      </div>
    </div>
    <div
      class="grid cols2 grid"
    >
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
          for="postalCode"
        >
          Postinumero
          <span
            aria-hidden="true"
            class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
          >
            *
          </span>
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="postalCode"
            required=""
            type="text"
            value="00170"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
          for="city"
        >
          Toimipaikka
          <span
            aria-hidden="true"
            class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
          >
            *
          </span>
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="city"
            required=""
            type="text"
            value="Helsinki"
          />
        </div>
      </div>
    </div>
    <hr
      class="divider"
    />
    <span
      class="text standard span m bold"
    >
      Yhteyshenkilö
    </span>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="firstName"
      >
        Etunimet
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="firstName"
          required=""
          type="text"
          value="Lauri Kristian"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="lastName"
      >
        Sukunimi
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="lastName"
          required=""
          type="text"
          value="Relander"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="phone"
      >
        Puhelinnumero
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="phone"
          type="text"
          value="+358 00 000 0000"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="email"
      >
        Sähköpostiosoite
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="email"
          required=""
          type="text"
          value="test@example.com"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="comment"
      >
        Huomiot
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <textarea
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="comment"
        />
      </div>
    </div>
    <div
      class="formActionButtons"
    >
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        color="supplementary"
        type="button"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Peruuta
        </span>
      </button>
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_primary__3bo_T button_hds-button--primary__2NVvO Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="submit"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Tallenna
        </span>
      </button>
    </div>
  </form>,
]
`;

exports[`CustomerForm renders correctly with private customer profile 1`] = `
Array [
  <h4
    class="text brand h4"
  >
    ASIAKKAAN TIEDOT
  </h4>,
  <form
    action="#"
    class="form"
  >
    <div
      class="Select-module_root__Ka5uO select"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="customerGroup-toggle-button"
        id="customerGroup-label"
      >
        Asiakasryhmä
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="Select-module_wrapper__1WQXs"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="customerGroup-label customerGroup-toggle-button"
          aria-owns="customerGroup-menu"
          class="Select-module_button__1aIsm"
          id="customerGroup-toggle-button"
          type="button"
        >
          Yksityinen
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
        <ul
          aria-labelledby="customerGroup-label"
          aria-required="true"
          class="Select-module_menu__1H2aU"
          id="customerGroup-menu"
          role="listbox"
          style="max-height: 260px;"
          tabindex="-1"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="firstName"
      >
        Etunimet
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="firstName"
          required=""
          type="text"
          value="Pehr Evind"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="lastName"
      >
        Sukunimi
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="lastName"
          required=""
          type="text"
          value="Svinhufvud"
        />
      </div>
    </div>
    <hr
      class="divider"
    />
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="address"
      >
        Jakeluosoite
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="address"
          required=""
          type="text"
          value="Mariankatu 2"
        />
      </div>
    </div>
    <div
      class="grid cols2 grid"
    >
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
          for="postalCode"
        >
          Postinumero
          <span
            aria-hidden="true"
            class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
          >
            *
          </span>
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="postalCode"
            required=""
            type="text"
            value="00170"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
          for="city"
        >
          Toimipaikka
          <span
            aria-hidden="true"
            class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
          >
            *
          </span>
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="city"
            required=""
            type="text"
            value="Helsinki"
          />
        </div>
      </div>
    </div>
    <hr
      class="divider"
    />
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="phone"
      >
        Puhelinnumero
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="phone"
          type="text"
          value="+358 00 000 0000"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="email"
      >
        Sähköpostiosoite
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="email"
          required=""
          type="text"
          value="test@example.com"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="comment"
      >
        Huomiot
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <textarea
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="comment"
        />
      </div>
    </div>
    <div
      class="formActionButtons"
    >
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        color="supplementary"
        type="button"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Peruuta
        </span>
      </button>
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_primary__3bo_T button_hds-button--primary__2NVvO Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="submit"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Tallenna
        </span>
      </button>
    </div>
  </form>,
]
`;
